zenodo/modules/deposit/views.py

```diff
@@ -259,6 +260,18 @@ def current_datetime():
         'current_time': now.time(),
     }
 
+@blueprint.route(
+    '/record/extractmetadata/<key>/<version_id>',
+    methods=['POST']
+)
+def extractmetadata(key=None, version_id=None):
```

Comment:
key is not necessary if you have ``version_id`` (one of PKs of ObjectVersion model). If we need the key later we can always get it: https://github.com/inveniosoftware/invenio-files-rest/blob/master/invenio_files_rest/models.py#L801